app/controllers/reservations_controller.rb

```diff
@@ -183,6 +183,15 @@ def update
       return
     end
 
+    if !current_user.administrator?
```

Comment:
In the context where this action is used, is this check necessary? Most of the time, won't the administrator will be using one of the other reservation change views (like the order detail edit popup).